tests/testthat/test-unnest.R

```diff
@@ -0,0 +1,313 @@
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(tags),
+    as.data.frame(df) |> tidyr::unnest_longer(tags) |> as.data.frame()
+  )
+})
+
+test_that("column can be specified as string", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    values = list(c(1, 2), c(3, 4))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars("values"),
+    as.data.frame(df) |> tidyr::unnest_longer("values") |> as.data.frame()
+  )
+})
+
+test_that("multiple columns can be unnested together", {
+  df <- polars::pl$DataFrame(
+    id = 1:3,
+    a = list(c(1, 2), c(3, 4), c(5, 7, 6)),
+    b = list(c("x", "y"), c("z", "w"), c("u", "v", "w"))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(c(a, b)),
+    as.data.frame(df) |> tidyr::unnest_longer(c(a, b)) |> as.data.frame()
+  )
+})
+
+test_that("chained unnest_longer works", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    a = list(1:2, 3:4),
+    b = list(5:6, 7:8)
+  )
+
+  expect_equal(
+    df |>
+      unnest_longer_polars(a) |>
+      unnest_longer_polars(b),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(a) |>
+      tidyr::unnest_longer(b) |>
+      as.data.frame()
+  )
+})
+
+test_that("multiple columns can be specified as character vector", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    a = list(c(1, 2), c(3, 4)),
+    b = list(c("x", "y"), c("z", "w"))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(c("a", "b")),
+    as.data.frame(df) |> tidyr::unnest_longer(c("a", "b")) |> as.data.frame()
+  )
+})
+
+test_that("values_to errors with multiple columns without template", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    a = list(1:2, 3:4),
+    b = list(5:6, 7:8)
+  )
+
+  expect_snapshot(
+    df |> unnest_longer_polars(c(a, b), values_to = "val"),
+    error = TRUE
+  )
+})
+
+test_that("indices_to errors with multiple columns without template", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    a = list(1:2, 3:4),
+    b = list(5:6, 7:8)
+  )
+
+  expect_snapshot(
+    df |> unnest_longer_polars(c(a, b), indices_to = "idx"),
+    error = TRUE
+  )
+})
+
+test_that("values_to template works with multiple columns", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(c(y, z), values_to = "{col}_val"),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        values_to = "{col}_val"
+      ) |>
+      as.data.frame()
+  )
+})
+
+test_that("indices_to template works with multiple columns", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  # Warning: `tidyr::unnest_longer()` places the index column immediately
+  # after the corresponding value column.
+  # `unnest_longer_polars()` currently places all index columns at the end.
+  columns <- c("x", "y", "y_idx", "z", "z_idx")
```

Comment:
This bothers me a bit, is it possible to rearrange columns so that the output order is the same as in `tidyr`?